src/plots/plots.js

```diff
@@ -864,8 +868,57 @@ plots.linkSubplots = function(newFullData, newFullLayout, oldFullData, oldFullLa
             null :
             axisIDs.getFromId(mockGd, ax.anchor);
     }
+
+    // finally, we can find the main subplot for each axis
+    // (on which the ticks & labels are drawn)
+    for(i = 0; i < axList.length; i++) {
+        ax = axList[i];
+        ax._mainSubplot = findMainSubplot(ax, newFullLayout);
+    }
 };
 
+function findMainSubplot(ax, fullLayout) {
```

Comment:
Is moving `findMainSubplot` from `lsInner` to `supplyDefaults` necessary? I suspect this will make `supplyDefaults` for large splom traces significantly slower.

Reply:
Something or other failed with it in `lsInner`, though I don't recall what. I'll take a look at 🐎 for large sploms, I'm sure there's a shortcut we can take there.

Reply:
> I'm sure there's a shortcut we can take there.

Yeah, I would be nice to leave `findMainSubplot` in `supplyDefaults` and add a `fullLayout._hasOnlyLargeSploms` condition to bypass the loop-over-subplot-for-all-axes loops.

Reply:
Large sploms usually avoid the subplot loop there - the [first try](https://github.com/plotly/plotly.js/pull/3323/files/858f7c2aea73f03f06215c280dd11c3429d11584..bdabf48c298001f490368618db0bd4281ae22a89#diff-ad4f76ccd6044ed16514297078e13b84R894) based on `_anchorAxis` works unless the labels are unanchored - upper half with lower labels, or both halves with no diagonal.

Anyway I added an extra collection of counteraxes and subplots *per axis* in https://github.com/plotly/plotly.js/pull/3323/commits/f55e7695209a235ced67c27d39ef4055782b93c4, which means we don't have to search the full subplot list anymore here (and in fact `Axes.getSubplots` is completely unused within plotly.js, but we still have it in streambed ATM). So even when splom *does* go through this loop, it now only takes 0.3ms at 20 dimensions - ~2ms at 50 dimensions if it's O(n^2).

Reply:
> works unless the labels are unanchored - upper half with lower labels, or both halves with no diagonal.

Ha right, I remember making that happen 3 or 4 splom-perf PRs ago. 

> So even when splom _does_ go through this loop, it now only takes 0.3ms at 20 dimensions - ~2ms at 50 dimensions if it's O(n^2).

Beautiful :racehorse: